kura/org.eclipse.kura.core.configuration/src/main/java/org/eclipse/kura/core/configuration/ConfigurationServiceImpl.java

```diff
@@ -678,7 +682,12 @@ synchronized void unregisterComponentConfiguration(String pid) {
 
     boolean mergeWithDefaults(OCD ocd, Map<String, Object> properties) throws KuraException {
         boolean changed = false;
-        Set<String> keys = properties.keySet();
+        Set<String> keys;
```

Comment:
I don't think it makes sense to call this method with null properties. The method modifies the instance referenced by the properties parameter :-( so I would consider this occurrence a bug and an NPE or any runtime exception should be appropriate.